src/mappings/token.ts

```diff
@@ -21,14 +17,25 @@ export function fetchTokenDetails(event: ethereum.Event): Token | null {
   //check if token details are already saved
   let token = Token.load(event.address.toHex());
   if (!token) {
-    //if token details are not available
-    //create a new token
+    // If token details are not available, create a new token
     token = new Token(event.address.toHex());
 
+    //set the address field
+    token.address = event.address;
+
     //set some default values
     token.name = "N/A";
     token.symbol = "N/A";
     token.decimals = 0;
+    token.eventCount = ZERO; // Initialize eventCount
+    token.burnEventCount = ZERO; // Initialize eventCount
+    token.mintEventCount = ZERO; // Initialize eventCount
+    token.transferEventCount = ZERO;
+    token.holderCount = ZERO;
+    token.totalSupply = BigDecimal.fromString('0');
```

Comment:
If `totalSupply` has default value, this seems unnecessary

Reply:
Besides, it has initialization code a few lines below

Reply:
totalSupply change according to the burn and mint events.